components/skus/browser/skus_service.cc

```diff
@@ -0,0 +1,152 @@
+    std::move(callback_state->cb).Run("");
+  }
+  delete callback_state;
+}
+
+void OnPrepareCredentialsPresentation(
+    skus::PrepareCredentialsPresentationCallbackState* callback_state,
+    skus::SkusResult result,
+    rust::cxxbridge1::Str presentation) {
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run(static_cast<std::string>(presentation));
+  }
+  delete callback_state;
+}
+
+void OnCredentialSummary(skus::CredentialSummaryCallbackState* callback_state,
+                         skus::SkusResult result,
+                         rust::cxxbridge1::Str summary) {
+  std::string summary_string = static_cast<std::string>(summary);
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          summary_string, base::JSONParserOptions::JSON_PARSE_RFC);
+  absl::optional<base::Value>& records_v = value_with_error.value;
+
+  if (records_v && callback_state->prefs) {
+    if (callback_state->domain == "vpn.brave.com" ||
+        callback_state->domain == "vpn.bravesoftware.com" ||
+        callback_state->domain == "vpn.brave.software") {
+      const base::Value* active = records_v->FindKey("active");
+      if (active) {
+        bool has_credential = active && active->is_bool() && active->GetBool();
+        callback_state->prefs->SetBoolean(skus::prefs::kSkusVPNHasCredential,
+                                          has_credential);
+      }
+    }
+  }
+
+  if (callback_state->cb) {
+    std::move(callback_state->cb).Run(summary_string);
+  }
+  delete callback_state;
+}
+
+}  // namespace
+
+namespace skus {
+
+SkusService::SkusService(
+    PrefService* prefs,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : context_(
+          std::make_unique<skus::SkusContextImpl>(prefs, url_loader_factory)),
+      sdk_(initialize_sdk(std::move(context_), skus::GetEnvironment())),
+      prefs_(prefs) {}
+
+SkusService::~SkusService() {}
+
+mojo::PendingRemote<mojom::SkusService> SkusService::MakeRemote() {
+  mojo::PendingRemote<mojom::SkusService> remote;
+  receivers_.Add(this, remote.InitWithNewPipeAndPassReceiver());
+  return remote;
+}
+
+void SkusService::Bind(mojo::PendingReceiver<mojom::SkusService> receiver) {
+  receivers_.Add(this, std::move(receiver));
+}
+
+void SkusService::RefreshOrder(
+    const std::string& order_id,
+    mojom::SkusService::RefreshOrderCallback callback) {
+  std::unique_ptr<skus::RefreshOrderCallbackState> cbs(
+      new skus::RefreshOrderCallbackState);
+  cbs->cb = std::move(callback);
+
+  sdk_->refresh_order(OnRefreshOrder, std::move(cbs), order_id);
+}
+
+void SkusService::FetchOrderCredentials(
+    const std::string& order_id,
+    mojom::SkusService::FetchOrderCredentialsCallback callback) {
+  std::unique_ptr<skus::FetchOrderCredentialsCallbackState> cbs(
+      new skus::FetchOrderCredentialsCallbackState);
+  cbs->cb = std::move(callback);
+  cbs->order_id = order_id;
```

Comment:
What is this for?

Reply:
`fetch_order_credentials` is called after `refresh_order`. Both require the order ID which account.brave.com is providing

Reply:
sorry, I should have been more specific because I was asking why we were adding extra data to FetchOrderCredentialsCallbackState instead of just binding order_id to the callback

Reply:
Removed in https://github.com/brave/brave-core/pull/10847/commits/918ee0a11f6e9b1653d7edde1a713d3eca98a9f8